openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +199,12 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        if "account_privacy" in update:
+            if account_privacy is not None:
+                account_privacy_serializer.save()
+            else:
+                delete_user_preference(requesting_user, "account_privacy")
+
```

Comment:
Please add a comment to this effect (above the call to `update_user_preferences`) so other developers who run into this realize why validating&saving together is Ok in this function.  Besides, as new data is rolled into this API, we need to find a design that allows decoupling of the code and non-duplication.  So the wrapping transaction allows this.